src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -197,8 +197,10 @@ private static SyntaxNode WithBlockBody(SyntaxNode node, BlockSyntax body)
         {
             declarationStatement = declarationStatement.WithAdditionalAnnotations(Formatter.Annotation);
 
-            var oldOutermostBlock = block;
-            var matches = FindMatches(document, expression, document, oldOutermostBlock, allOccurrences, cancellationToken);
+            // If we're within a local function, our scope for the new local declaration is expanded to include the enclosing member.
+            var scope = block.GetAncestor<LocalFunctionStatementSyntax>() == null ? block : (SyntaxNode)block.GetAncestor<MemberDeclarationSyntax>();
```

Comment:
If it isn't especially expensive, would it be simpler to just always look for the `MemberDeclarationSyntax`?

```
var scope = (SyntaxNode)block.GetAncestorOrSelf<MemberDeclarationSyntax>();
```

Reply:
I think you're correct. Unfortunately I didn't realize that we had to ensure the local function was non-static, so I think the check is unavoidable now.